### PR TITLE
Refactor deploy:cleanup Command

### PR DESCRIPTION
**Pull request by amayer5125**
Remove $runOpts since it's not actually used
Move release symlink deletion to beginning of method. This will allow the release symlink to be deleted even if deleting the old releases fails.

### CODE REVIEW

**Comment by Schrank** [comment]
Thanks for the PR! @antonmedv does it make sense to delete everything, EXCEPT the last five releases? (making sure the list of the kept releases has really 5 unique entries?

https://github.com/deployphp/deployer/issues/2759

File: recipe/deploy/cleanup.php [hunk 1]
```diff
@@ -4,26 +4,17 @@
 // Use sudo in deploy:cleanup task for rm command.
 set('cleanup_use_sudo', false);
 
-desc('Cleanups old releases');
+desc('Cleanup old releases');
```
Comment (antonmedv) [inline recipe/deploy/cleanup.php:7]: Let’s keep an s at the end. To be in descriptive form.

**Comment by antonmedv** [comment]
@Schrank yes, I think this makes sense. Can @amayer5125 you do it as well in this pr? More context it #2759

File: recipe/deploy/cleanup.php (hunk 1 above)
Comment (peterjaap) [reply recipe/deploy/cleanup.php:7]: But that's not correct English? You probably mean "Cleans up old releases"

Comment (antonmedv) [reply recipe/deploy/cleanup.php:7]: Yes! I’m not native English bearer and will need proofread of docs after I finish. 😌

File: recipe/deploy/release.php [hunk 2]
```diff
@@ -18,21 +18,13 @@
...
-    $releasesLog = [];
-    foreach (explode("\n", $data) as $line) {
-        $metainfo = json_decode($line, true);
-        if (!empty($metainfo)) {
-            $releasesLog[] = $metainfo;
-        }
-    }
-    return $releasesLog;
+    return array_filter($releaseLogs, function ($release) {
+        return ! empty($release);
```
Comment (Schrank) [inline recipe/deploy/release.php:26]: No need for this (I guess?) because array_filter by defaults removes everything which evaluates to false

**Comment by Schrank** [comment]
I like all the array functions, but most people are not used to them. Can you please comment  them? 😅

**Comment by amayer5125** [comment]
@Schrank I added comments to the `array_*` functions.

File: recipe/deploy/release.php (hunk 2 above)
Comment (amayer5125) [reply recipe/deploy/release.php]: Updated as suggested.

File: recipe/deploy/release.php [hunk 3]
```diff
@@ -44,33 +36,31 @@
         return [];
     }
 
-    // Will list only dirs in releases.
-    $ll = explode("\n", run('cd releases && ls -t -1 -d */'));
-    $ll = array_map(function ($release) {
+    // retrieve all release directories
```
Comment (antonmedv) [inline recipe/deploy/release.php:39]: Why this change here? Let’s separate into two PR.

**Review by antonmedv — CHANGES_REQUESTED** [review]

File: recipe/deploy/release.php (hunk 3 above)
Comment (amayer5125) [reply recipe/deploy/release.php:39]: > @Schrank yes, I think this makes sense. Can @amayer5125 you do it as well in this pr? More context it #2759

Didn't you ask me to look at it here? Maybe I didn't need to refactor releases_list, but just releases_log to fix the issue? I'll separate it out into another PR.

File: recipe/deploy/cleanup.php (hunk 1 above)
Comment (amayer5125) [reply recipe/deploy/cleanup.php:7]: Is this okay the way it is, or did you still want me to revert it?

File: recipe/deploy/release.php (hunk 3 above)
Comment (amayer5125) [reply recipe/deploy/release.php]: I moved the changes you mentioned to another PR (#2845).

Comment (antonmedv) [reply recipe/deploy/release.php]: Thanks, got out of context. Will review it today. 

Thanks for your contributions!
